eth_utils/numeric.py

```diff
@@ -12,13 +14,16 @@ def __gt__(self, other: Any) -> bool:
         ...
 
 
-TValue = TypeVar("TValue", bound=Comparable)
+TComparable = Union[Comparable, numbers.Real, int, float, decimal.Decimal]
+
+
+TValue = TypeVar("TValue", bound=TComparable)
 
 
 def clamp(lower_bound: TValue, upper_bound: TValue, value: TValue) -> TValue:
-    if value < lower_bound:
+    if value < lower_bound:  # type: ignore
```

Comment:
weird, why does this need a type ignore?

Reply:
I'll add a comment.  The error that `mypy` gave was not helpful, just that the comparison operator was not valid.

Reply:
I think it is not valid because you can not compare `Union` types with each other which is what the code tries to do. The bound is `TCompareable` and `TCompareable` is an alias to `Union[Comparable, numbers.Real, int, float, decimal.Decimal]`.

It seems we could add a `Protocol` and use that as the bound:

https://stackoverflow.com/questions/47965083/comparable-types-with-mypy

Another option might be to use function overloading.

https://mypy.readthedocs.io/en/latest/more_types.html#function-overloading

Reply:
I'm inclined to add a comment and keep the ignores as I don't really see what the overloading gives us other than a more complicated looking module. Thoughts?

Reply:
Fine by me